capacity4more/modules/c4m/message/c4m_message/c4m_message.module

```diff
@@ -881,13 +878,38 @@ function c4m_message_load_homepage_activity_stream() {
     'csrfToken' => drupal_get_token(\RestfulInterface::TOKEN_VALUE),
     'activities' => $activities,
     'data' => array(
-      'entity' => array('group' => isset($groups['node']) ? $groups['node'] : $groups),
+      'entity' => array('group' => $groups),
     ),
   );
   drupal_add_js($settings, 'setting');
 }
 
 /**
+ * Implementation of hook_query_TAG_alter().
+ *
+ * Changes the join of the group field to "LEFT" join,
+ * So it's possible to get messages without this field,
+ * Also changes the condition from AND to OR. (field group has those values OR is null).
+ */
+function c4m_message_query_activity_stream_homepage_alter(SelectQuery &$query) {
+  $tables = &$query->getTables();
+  $conditions = &$query->conditions();
+
+  $tables['field_data_field_group_node1']['join type'] = 'LEFT';
+  $values = $conditions[1]['value'];
+  $or = db_or()
+    ->condition('field_group_node_target_id',  $values, 'IN')
+    ->isNull('field_group_node_target_id');
+   $query->condition($or);
+
+  foreach ($conditions as $key => $condition) {
+    if ($conditions['field'] == 'field_data_field_group_node1.field_group_node_target_id') {
+      unset($conditions[$key]);
```

Comment:
This looks like we are removing the condition that we've just added in the db_or.